### PR TITLE
Don't prepend ./ to file paths' when using --dir=.

### PR DESCRIPTION
**Pull request by jez**
### Motivation

This makes some things easier in the Sorbet Compiler. It also looks better in
error messages.

### Test plan

See included automated tests.

### CODE REVIEW

File: common/common.cc [hunk 1]
```diff
@@ -248,6 +248,8 @@ bool sorbet::FileOps::isFileIgnored(string_view basePath, string_view filePath,
     return false;
 }
 
+namespace {
+
```
Comment (elliottt) [inline common/common.cc:252]: 🎉

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KKZRmzef7ODrna
→ https://go/builds/bui_KKZSHZ7G4EVt10

File: common/common.cc [hunk 2]
```diff
@@ -279,22 +282,30 @@ void appendFilesInDir(string_view basePath, const string &path, const sorbet::Un
             if (dotLocation != string::npos) {
                 string_view ext(fullPath.c_str() + dotLocation, fullPath.size() - dotLocation);
                 if (extensions.contains(ext)) {
-                    result.push_back(move(fullPath));
+                    // Remove the ./ prefix in the final path (but it's convenient to have it
+                    // above for the sake of matching ignore patterns). Also don't allocate unless we have to.
+                    if (basePath == ".") {
+                        result.push_back(fullPath.substr(2));
```
Comment (elliottt) [inline common/common.cc:288]: What about using `fullPath.erase(0, 2)` instead of copying to make the substring?

File: packager/packager.cc [hunk 3]
```diff
@@ -182,6 +182,11 @@ class PackageDB final {
             curPrefixPos = path.find_last_of('/', curPrefixPos - 1);
         }
 
+        const auto &root = packageInfoByPathPrefix.find("");
```
Comment (elliottt) [inline packager/packager.cc:185]: This returns an iterator, I'm not sure it needs the `&`.
```suggestion
        const auto root = packageInfoByPathPrefix.find("");
```

**Review by elliottt — APPROVED** [review]
Looks good, just a few comments 👍

**Pull request merged**